### PR TITLE
Fix serialization of pod specs in pod plugin

### PR DESCRIPTION
**Pull request by jeevb**
Signed-off-by: Jeev B <jeev.balakrishnan@freenome.com>

# TL;DR
Pod specs in the `custom` struct of `TaskTemplate` were being improperly serialized, resulting in a subsequent parsing failure by Flytepropeller.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: plugins/tests/pod/test_pod.py [hunk 1]
```diff
@@ -81,6 +87,57 @@ def simple_pod_task(i: int):
     assert config["primary_container_name"] == "a container"
 
 
+def test_pod_task():
```
Comment (jeevb) [inline plugins/tests/pod/test_pod.py:90]: Not sure why github renders like this This is exactly the same test as before with the underscores changed to camelcase.

File: plugins/tests/pod/test_pod.py [hunk 2]
```diff
@@ -19,7 +22,7 @@ def get_pod_spec():
     return pod_spec
 
 
-def test_pod_task():
+def test_pod_task_deserialization():
```
Comment (jeevb) [inline plugins/tests/pod/test_pod.py:25]: New test that actually deserializes the jsonified `custom` dict into a `V1PodSpec` object to ensure that serialization is correct.

**Review by katrogan — APPROVED** [review]
thank you!
